src/commands/xkcd.ts

```diff
@@ -0,0 +1,165 @@
+import {
+	ApplicationCommandOptionType,
+	EmbedBuilder,
+	InteractionReplyOptions,
+	MessageReplyOptions,
+} from 'discord.js';
+import axios from 'axios';
+
+// defining the response
+interface GetComicResponse {
+	month: string;
+	num: number;
+	link: string;
+	news: string;
+	safe_title: string;
+	transcript: string;
+	alt: string;
+	year: string;
+	title: string;
+	day: string;
+	img: string;
+}
+
+// a quick call to see the latest comic.
+// @return the data of the most recent comic, or an empty response with num set to -1 to indicate an error.
+async function _latestCheck(): Promise<GetComicResponse> {
+	try {
+		const { data, status } = await axios.get<GetComicResponse>('https://xkcd.now.sh/?comic=latest');
+		if (status !== 200) throw new Error('No good status');
+		return data;
+	} catch (error_) {
+		console.log(error_);
+		const error: GetComicResponse = {
+			month: '',
+			num: -1,
+			link: '',
+			news: '',
+			safe_title: '',
+			transcript: '',
+			alt: '',
+			year: '',
+			title: '',
+			day: '',
+			img: '',
+		};
+		return error;
+	}
+}
+
+// this is kinda cursed, but its the best way that I can pass "reply" as a parameter to minimize
+// code duplication
```

Comment:
Have you considered defining this function inside of the `execute` function body? That way, you can reference `reply` directly, without needing to copy/paste its type here.

Reply:
Dang, I'm still in classic C mind. That's a much more elegant solution lol.